auth0/resource_auth0_user.go

```diff
@@ -0,0 +1,163 @@
+package auth0
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/yieldr/go-auth0/management"
+)
+
+func newUser() *schema.Resource {
+	return &schema.Resource{
+		Create: createUser,
+		Read:   readUser,
+		Update: updateUser,
+		Delete: deleteUser,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				StateFunc: func(val interface{}) string {
+					return "auth0|" + val.(string)
+				},
+			},
+			"conn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"email": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"username": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"password": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"phone_number": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"user_metadata": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+			"app_metadata": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+			"email_verified": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"verify_email": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"phone_verified": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func readUser(d *schema.ResourceData, m interface{}) error {
+	api := m.(*management.Management)
+	u, err := api.User.Read(d.Id())
+	if err != nil {
+		return err
+	}
+
+	for k, v := range map[string]interface{}{
+		"user_id":        u.ID,
+		"conn":           u.Connection,
+		"username":       u.Username,
+		"phone_number":   u.PhoneNumber,
+		"user_metadata":  u.UserMetadata,
+		"email_verified": u.EmailVerified,
+		"phone_verified": u.PhoneVerified,
+		"verify_email":   u.VerifyEmail,
+		"app_metadata":   u.AppMetadata,
+		"email":          u.Email,
+		"password":       u.Password,
+	} {
+		if (k == "password" || k == "conn") && v.(*string) == nil {
+			continue
+		}
+
+		if err := d.Set(k, v); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func createUser(d *schema.ResourceData, m interface{}) error {
+	u := buildUser(d)
+	api := m.(*management.Management)
+	if err := api.User.Create(u); err != nil {
+		return err
+	}
+	d.SetId(*u.ID)
+	return nil
+}
+
+func updateUser(d *schema.ResourceData, m interface{}) error {
+	api := m.(*management.Management)
+
+	for k, v := range map[string]*management.User{
+		"username":       {Username: String(d, "username"), Connection: String(d, "conn")},
+		"password":       {Password: String(d, "password"), Connection: String(d, "conn")},
+		"email":          {Email: String(d, "email"), Connection: String(d, "conn")},
+		"email_verified": {EmailVerified: Bool(d, "email_verified"), Connection: String(d, "conn")},
+	} {
+		if d.HasChange(k) {
```

Comment:
It was to get ride of this bunch of error :
<img width="454" alt="capture d ecran 2018-10-21 a 01 15 42" src="https://user-images.githubusercontent.com/4854264/47261403-027f0e00-d4cf-11e8-9fa5-0665c7c81eba.png">

Reply:
Then maybe a good application for [`ConflictsWith`](https://github.com/yieldr/terraform-provider-auth0/blob/d0a2ce6d27a0fecb734f588ca945636288692981/vendor/github.com/hashicorp/terraform/helper/schema/schema.go#L170)?

Reply:
I don't really get how you use that and for which purpose exactly. But you still have to update partially if you want to have a correct overall behaviour.

Reply:
`ConflictsWith` would disallow specifying one if the other is specified. For example:

```
"email": {
  Type:     schema.TypeString,
  Optional: true,
  ConflictsWith: []string{"password"},
},
"password": {
  Type:     schema.TypeString,
  Optional: true,
  ConflictsWith: []string{"email"},
}
```

Would not allow you to set `password` if `email` is set.

To see if I get this right, is it allowed to specify `username` and `email` but not at the same time?

Reply:
You can't change in the same time username, email, email_verified and password. I think I get the point but in my opinion, we don't want this behaviour, we want to be able to define everything in one go, because you want to be able for instance to define a username and an email in your terraform and having changes reflected in one apply without having to mess around with your config, you want everything to be managed in the background.